stl/inc/generator

```diff
@@ -0,0 +1,600 @@
+                generator<_Rty, _Vty, _Alloc> _Gen;
+
+                explicit _Awaitable(generator<_Rty, _Vty, _Alloc>&& _Gen_) noexcept : _Gen(_STD move(_Gen_)) {}
+
+                _NODISCARD bool await_ready() noexcept {
+                    return !_Gen._Coro;
+                }
+
+                template <class _CoroPromise>
+                _NODISCARD coroutine_handle<_Promise_base> await_suspend(
+                    coroutine_handle<_CoroPromise> _Current) noexcept {
+                    // Push _Gen's coroutine onto the coroutine stack that _Current is at the top of.
+#ifdef __cpp_lib_is_pointer_interconvertible // TRANSITION, LLVM-48860
+                    _STL_INTERNAL_STATIC_ASSERT(is_pointer_interconvertible_base_of_v<_Promise_base, _CoroPromise>);
+#endif // ^^^ no workaround ^^^
+                    auto _Target    = coroutine_handle<_Promise_base>::from_address(_Gen._Coro.address());
+                    _Nested._Parent = coroutine_handle<_Promise_base>::from_address(_Current.address());
+                    if (const auto _Parent_nest_info = _Nested._Parent.promise()._Try_get_nest_info()) {
+                        _Nested._Root = _Parent_nest_info->_Root;
+                    } else {
+                        _Nested._Root = _Nested._Parent;
+                    }
+                    _Nested._Root.promise()._Set_top(_Target);
+                    _Target.promise()._Set_nest_info(_STD addressof(_Nested));
+                    return _Target;
+                }
+
+                void await_resume() {
+                    if (_Nested._Except) {
+                        _STD rethrow_exception(_STD move(_Nested._Except));
+                    }
+                }
+            };
+        };
+
+        _NODISCARD _Nest_info* _Try_get_nest_info() const noexcept {
+            if ((_Data & 1U) != 0) {
+                return reinterpret_cast<_Nest_info*>(_Data ^ 1U);
+            }
+
+            return nullptr;
+        }
+
+        _NODISCARD coroutine_handle<_Promise_base> _Get_top() const noexcept {
+            _STL_INTERNAL_CHECK((_Data & 1U) == 0);
+            return coroutine_handle<_Promise_base>::from_address(reinterpret_cast<void*>(_Data));
+        }
+
+        void _Set_nest_info(_Nest_info* _Info) noexcept {
+            _Data = reinterpret_cast<uintptr_t>(_Info) | 1U;
+        }
+
+        void _Set_top(coroutine_handle<_Promise_base> _Top) noexcept {
+            _Data = reinterpret_cast<uintptr_t>(_Top.address());
+        }
+
+        template <class, class>
+        friend struct _Iter_provider;
+
+        // Least significant bit of `_Data` indicates stored information:
+        // LSB 0: `_Data` is a top coroutine handle,
+        // LSB 1: `_Data ^ 1U` is a pointer to an object of type `_Nest_info`.
+        uintptr_t _Data = reinterpret_cast<uintptr_t>(coroutine_handle<_Promise_base>::from_promise(*this).address());
+        add_pointer_t<_Yielded> _Ptr = nullptr;
+    };
+
+    struct _Secret_tag {};

```

Comment:
We should have an explicit default ctor to make this moderately difficult to misuse, right?